src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs

```diff
@@ -0,0 +1,199 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.MachineLearningCompute.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.MachineLearningCompute;
+using Microsoft.Azure.Management.MachineLearningCompute.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.MachineLearningCompute.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Set, CmdletSuffix, SupportsShouldProcess = true, DefaultParameterSetName = CmdletParametersParameterSet)]
+    [OutputType(typeof(PSOperationalizationCluster))]
+    public class SetAzureRmMlOpCluster : MachineLearningComputeCmdletBase
+    {
+        protected const string ObjectParameterSet = "SetByInputObject";
+
+        protected const string ResourceIdParameterSet = "SetByResourceId";
+
+        protected const string CmdletParametersParameterSet = "SetByIndividualParameters";
+
+        [Parameter(ParameterSetName = ObjectParameterSet,
+            Mandatory = true, 
+            ValueFromPipeline = true,
+            HelpMessage = ClusterParameterHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        [Alias(ClusterInputObjectAlias)]
+        public PSOperationalizationCluster InputObject { get; set; }
```

Comment:
@shutchings had a small internal discussion about what to do with `InputObject` in the piping scenario for `Set-*` cmdlets, and we decided that the `InputObject` parameter set should have all of the property parameters as the other sets, and if the user passes a value for any property parameter, then you update the corresponding property of the `InputObject` before passing it to the server.

Reply:
That makes sense. Check out the latest changes.